src/asound/bluealsa-pcm.c

```diff
@@ -564,8 +564,11 @@ static int bluealsa_hw_params(snd_pcm_ioplug_t *io, snd_pcm_hw_params_t *params)
 	if (!bluealsa_dbus_pcm_open(&pcm->dbus_ctx, pcm->ba_pcm.pcm_path,
 				&pcm->ba_pcm_fd, &pcm->ba_pcm_ctrl_fd, &err)) {
 		debug2("Couldn't open PCM: %s", err.message);
+		ret = -EIO;
+		if (strcmp(err.message, strerror(EBUSY)) == 0)
```

Comment:
I'm not sure whether string comparison will work here. The problem is that bluealsa daemon and ALSA application can run under different locales, so the strings will not match. I think that we need error code to errno mapping (and change error codes in `bluealsa_pcm_open()`.